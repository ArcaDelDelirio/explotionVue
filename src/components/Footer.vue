<script>
import store from '../store'
export default {
    name:'Footer',
    props:["users"],
    computed:{
        apiStatus: () => store.state.apiState,
        responseTime: () => store.state.responseTime
    }
}
</script>


<template>
    <footer>
        <div class="flag-api">
            <div :class="{active:apiStatus}" class="status">
                <i><font-awesome-icon icon="circle"/></i>
            </div>
            {{responseTime}}ms
            <i> <font-awesome-icon icon="server"/> API State </i>
        </div>
        <div v-bind:key="user.id" v-for="user in users">
            <i v-if="user.id == 1">User: {{user.name}}</i> 
        </div>
    </footer>
</template>


<style scoped>
    footer{
        display: flex;
        bottom: 0;
    }
    .status{
        color:red;
    }
    .active{
        color:green;
    }
    .flag-api{
        display: flex;
    }
    ul{
       list-style-type: none; 
    }
</style>
