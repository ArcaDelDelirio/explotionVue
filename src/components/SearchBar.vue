<script>
import DataGallery from './Gallery';
export default {
  name: 'SearchBar',
  components: {
    DataGallery,
  },
  data() {
    return {
      searchUser: '',
    };
  },
};
</script>


<template>
  <article class="container">
    <div class="main-title">
      <h1>Search on Database</h1>
    </div>
    <div class="search-container">
      <form class="search-form">
        <input
          class="search-input"
          type="text"
          placeholder="ex:Title"
          name="search"
          v-model="searchUser"
        />
        <button class="search-button">
          <i>
            <font-awesome-icon icon="search" />
          </i>
        </button>
      </form>
    </div>
    <div id="result">
      <DataGallery :searchQuery="searchUser" />
    </div>
  </article>
</template>

<style scoped>
.container {
  display: grid;
  grid-template-columns: 25% 50% 25%;
  grid-template-rows: 10% 5% 85%;
}
.main-title {
  grid-row: 1;
  grid-column: 2 / 3;
}
.search-container {
  display: flex;
  grid-column: 2 / 3;
  grid-row: 2;
}
.search-form {
  flex-grow: 1;
  display: flex;
}
.search-input {
  flex-grow: 2;
}
#result {
  grid-row: 3;
  grid-column: 2/3;
}
</style>
