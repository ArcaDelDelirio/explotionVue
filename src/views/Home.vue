<template>
  <div class="home">
    <navData />
    <searchData />
    <FooterData />
  </div>
</template>

<script>
// @ is an alias to /src
import FooterData from '@/components/FooterData.vue'
import SearchData from '@/components/SearchData.vue'
import NavData from '@/components/NavData.vue'

export default {
  name: 'home',
  components: {
    NavData,
    SearchData,
    FooterData
  }
}
</script>
<style scoped>
  .home{
    display: grid;
    grid-template-rows: 7vh 81vh 12vh;
    height:100%;
  }
</style>
