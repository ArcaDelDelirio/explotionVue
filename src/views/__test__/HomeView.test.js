import { shallowMount } from '@vue/test-utils';
import NavBar from '../../components/NavBar';
import SearchBar from '../../components/SearchBar';
import Footer from '../../components/Footer';
import Home from '../Home';

describe('Home View', () => {
  const build = () => {
    const wrapper = shallowMount(Home, {
      data: () => ({
        movie: {},
      }),
    });

    return {
      wrapper,
      navigationData: () => wrapper.find(NavBar),
      searchData: () => wrapper.find(SearchBar),
      footerData: () => wrapper.find(Footer),
    };
  };

  it('Render the component', () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  }),
    it('Render the NavData component', () => {
      const { navigationData, searchData, footerData } = build();

      expect(navigationData().exists()).toBe(true);
      expect(searchData().exists()).toBe(true);
      expect(footerData().exists()).toBe(true);
    });
  // it('Pass a binded title movie props to DataGallery component', () => {
  //   const { wrapper, dataGallery } = build();
  //   wrapper.setData({
  //     movie: {
  //       title: 'Matrix',
  //     },
  //   });

  //   expect(dataGallery().vm.movie).toBe(wrapper.vm.movie);
  // });
});
