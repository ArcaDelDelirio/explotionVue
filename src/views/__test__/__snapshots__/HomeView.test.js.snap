// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View Render the component 1`] = `
<div class="home">
  <navbar-stub></navbar-stub>
  <searchbar-stub></searchbar-stub>
  <footer-stub></footer-stub>
</div>
`;
